src/Tools/ExternalAccess/FSharp/Internal/Diagnostics/FSharpDocumentDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ExternalAccess.FSharp.Diagnostics;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Internal.Diagnostics
+{
+    [Shared]
+    [ExportLanguageService(typeof(FSharpDocumentDiagnosticAnalyzerService), LanguageNames.FSharp)]
+    internal class FSharpDocumentDiagnosticAnalyzerService : ILanguageService
+    {
+        private readonly IFSharpDocumentDiagnosticAnalyzer _analyzer;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public FSharpDocumentDiagnosticAnalyzerService(IFSharpDocumentDiagnosticAnalyzer analyzer)
+        {
+            _analyzer = analyzer;
+        }
+
+        public Task<ImmutableArray<Diagnostic>> AnalyzeSemanticsAsync(Document document, CancellationToken cancellationToken)
+        {
+            return _analyzer.AnalyzeSemanticsAsync(document, cancellationToken);
+        }
+
+        public Task<ImmutableArray<Diagnostic>> AnalyzeSyntaxAsync(Document document, CancellationToken cancellationToken)
+        {
+            return _analyzer.AnalyzeSyntaxAsync(document, cancellationToken);
+        }
+    }
+
+    [DiagnosticAnalyzer(LanguageNames.FSharp)]
+    internal class FSharpDocumentDiagnosticAnalyzer : DocumentDiagnosticAnalyzer, IBuiltInAnalyzer
```

Comment:
:bulb: `IBuiltInAnalyzer` should not be used

```suggestion
    internal class FSharpDocumentDiagnosticAnalyzer : DocumentDiagnosticAnalyzer
```